nss-day-cohort-44/rare-rest-api-malicous-vortices
# Kn view current user posts

# ISSUE TICKET 
## Description
wrote the View for Posts on server side, with working create, retrieve, update, destroy, and list methods.

## Type of change
- [x ] New feature (non-breaking change which adds functionality)

# Testing Instructions
`git fetch origin Kn-view-current-user-posts `
`git checkout Kn-view-current-user-posts ` 

User Instructions: 
on post man  at http://localhost:8000/posts do a get method
-you should see a list of all the posts

on post man  at http://localhost:8000/posts/id of any post  do a get method
you will see that specific post

on post man  at http://localhost:8000/posts do a post method and put the following data in the body
`{
        "user": 1,
        "categoryId": 1,
        "title": "testv2",
        "publicationDate": "2020-08-28T14:51:39.989000Z",
        "imageUrl": "https//test",
        "content": "Trying out the post",
        "approved": false
}`
then do a get and you will see the newly created post.

on post man  at http://localhost:8000/posts/ id of new post you made do a put method and put the following data in the body
`{
        "user": 1,
        "categoryId": 1,
        "title": "testv3",
        "publicationDate": "2020-08-28T14:51:39.989000Z",
        "imageUrl": "https//test",
        "content": "Trying out the post",
        "approved": false
}`
you should see the title change from testv2 to testv3

on post man  at http://localhost:8000/posts/id of any post  do a delete method
then do a get method at http://localhost:8000/posts and the post will be gone









# Checklist:
- [x ] My code follows the style guidelines of this project
- [ x] I have performed a self-review of my own code
- [ x] My changes generate no new errors
- [ x] I have added test instructions that prove my fix is effective or that my feature works
